perfkitbenchmarker/providers/aws/aws_network.py

```diff
@@ -871,5 +1036,539 @@ def _Delete(self):
+
+  #   describe-vpn-gateways
+  # [--filters <value>]
+  # [--vpn-gateway-ids <value>]
+  # [--dry-run | --no-dry-run]
+  # [--cli-input-json <value>]
+  # [--generate-cli-skeleton <value>]
+  def _Exists(self):
+    """Returns true if the vpn gateway exists."""
+    describe_cmd = util.AWS_PREFIX + [
+        'ec2',
+        'describe-vpn-gateways',
+        '--region=%s' % self.region,
+        #         '--region=%s' % self.region,
+        '--filter=Name=vpn-gateway-id,Values=%s' % self.id
+    ]
+    stdout, _ = util.IssueRetryableCommand(describe_cmd)
+    response = json.loads(stdout)
+    #     logging.log(logging.INFO, response)
+    vpn_gateways = response['VpnGateways']
+    state = vpn_gateways[0]['State']
+    #     logging.log(logging.INFO, 'state: %s' % state)
+    #     logging.log(logging.INFO, 'vpngw_len: '.join(str(len(vpn_gateways))))
+    assert len(vpn_gateways) < 2, 'Too many VPN gateways.'
+    return len(vpn_gateways) > 0 and state != 'deleted'
+
+  #   delete-vpn-gateway
+  # --vpn-gateway-id <value>
+  # [--dry-run | --no-dry-run]
+  # [--cli-input-json <value>]
+  # [--generate-cli-skeleton <value>]
+  def _Delete(self):
+    """Deletes the vpn gateway."""
+    delete_cmd = util.AWS_PREFIX + [
+        'ec2',
+        'delete-vpn-gateway',
+        '--region=%s' % self.region,
+        '--vpn-gateway-id=%s' % self.id
+    ]
     vm_util.IssueCommand(delete_cmd)
+
+  def Attach(self, vpc_id):
+    """Attaches the vpn gateway to the VPC."""
+    if not self.attached:
+      self.vpc_id = vpc_id
+      attach_cmd = util.AWS_PREFIX + [
+          'ec2',
+          'attach-vpn-gateway',
+          '--region=%s' % self.region,
+          '--vpn-gateway-id=%s' % self.id,
+          '--vpc-id=%s' % self.vpc_id
+      ]
+      util.IssueRetryableCommand(attach_cmd)
+      self.attached = True
+
+  def Detach(self):
+    """Detaches the vpn gateway from the VPC."""
+    if self.attached:
+      detach_cmd = util.AWS_PREFIX + [
+          'ec2',
+          'detach-vpn-gateway',
+          '--region=%s' % self.region,
+          '--vpn-gateway-id=%s' % self.id,
+          '--vpc-id=%s' % self.vpc_id
+      ]
+      util.IssueRetryableCommand(detach_cmd)
+      self.attached = False
+
+  def AllowRoutePropogation(self, route_table_id):
+    cmd = util.AWS_PREFIX + [
+        'ec2',
+        'enable-vgw-route-propagation',
+        '--region=%s' % self.region,
+        '--gateway-id=%s' % self.id,
+        '--route-table-id=%s' % route_table_id
+    ]
+    util.IssueRetryableCommand(cmd)
+
+  def GetDefaultSecurityGroup(self):
+    cmd = util.AWS_PREFIX + [
+        'ec2',
+        'describe-security-groups',
+        '--region=%s' % self.region
+    ]
+    response = vm_util.IssueCommand(cmd)
+    response = json.loads(response)
+    for sg in response['SecurityGroups']:
+      if sg['GroupName'] == 'default':
+        return sg['GroupId']
+
+
+class AwsVPNGW(network.BaseVpnGateway):
+  CLOUD = providers.AWS
```

Comment:
Can you move all the AWS VPN related code (both existing and new) into a new file
  aws_vpn_network.py
as that will make it a lot easier to diff.  We did that with https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/providers/aws/aws_placement_group.py which made it a lot easier to review.

Reply:
Fixed.